File: app/src/main/java/in/mvpstarter/sample/injection/component/FragmentComponent.java
package in.mvpstarter.sample.injection.component;

import dagger.Subcomponent;
import in.mvpstarter.sample.injection.PerFragment;
import in.mvpstarter.sample.injection.module.FragmentModule;

/**
 * This component inject dependencies to all Fragments across the application
 */
@PerFragment
@Subcomponent(modules = FragmentModule.class)
public interface FragmentComponent {

}

File: app/src/main/java/in/mvpstarter/sample/injection/component/FragmentComponent.kt
package `in`.mvpstarter.sample.injection.component

import `in`.mvpstarter.sample.injection.PerFragment
import `in`.mvpstarter.sample.injection.module.FragmentModule
import dagger.Subcomponent

/**
 * This component inject dependencies to all Fragments across the application
 */
@PerFragment
@Subcomponent(modules = arrayOf(FragmentModule::class))
interface FragmentComponent